velero-plugins/common/shared.go

```diff
@@ -225,4 +231,77 @@ func StringInSlice(a string, list []string) bool {
 
 func StringPtr(s string) *string {
 	return &s
-}
\ No newline at end of file
+}
+
+func GetBackupStorageLocationForBackup(uid types.UID, name string, namespace string) (*velero.BackupStorageLocation, error) {
+	b, err := GetBackup(uid, name, namespace)
+	if err != nil {
+		return nil, err
+	}
+	if b.Spec.StorageLocation == "" {
```

Comment:
I think this is redundant, velero will not trigger a backup operation without a valid BSL associated with it, the backup will go in `FailedValidation` phase.